### PR TITLE
Fixes #281: Topology now skips non-local potential targets correctly

### LINKED ISSUE #281: NEST Topology crashes for fixed fan-out on multiple MPI processes
The following code will work correctly if run on a single MPI process, but crashes reporting an `IllegalConnection` in `ConnectLayers_i_i_D` when run on two or more processes:
```
/layer << /rows 3 /columns 3 /elements /iaf_neuron >> CreateLayer def
layer layer << /connection_type /divergent /number_of_connections 1 >> ConnectLayers
```
The code works fine for `/convergent` connections and also if a connection probability is given instead of a fixed number of outgoing connections.

### PR DESCRIPTION
**Pull request by heplesser**
When creating divergent connections with fixed fan-out, topology would try to connect to targets even if they were not local. This is now fixed.

### CODE REVIEW

**Comment by heplesser** [comment]
@jougs @hannahbos Could you take a look?

**Comment by tammoippen** [comment]
@jhnnsnk Can you have a look into this PR? Thank you.

**Comment by hannahbos** [comment]
@heplesser thanks. Looks good to me +1.

**Comment by tammoippen** [comment]
@jhnnsnk Can you have a look into this PR? Thank you.

File: topology/connection_creator_impl.h [hunk 1]
```diff
@@ -739,18 +739,28 @@ ConnectionCreator::divergent_connect_( Layer< D >& source, Layer< D >& target )
...
         source_id, target_ptr, target_ptr->get_thread(), synapse_model_, d, w );
-      is_selected[ random_id ] = true;
     }
   }
 }
 
 } // namespace nest
 
-#endif
+#endif
```
Comment (jougs) [inline topology/connection_creator_impl.h]: Can you please re-add the newline at the end of the file?

**Comment by jhnnsnk** [comment]
@heplesser Just tested it, looks good to me. Sorry for the late response.
:+1:

**Pull request merged**